stempy/reader.h

```diff
@@ -39,15 +39,78 @@ class StreamReader {
 
 public:
   StreamReader(const std::string &path, uint8_t version=1);
+  StreamReader(const std::vector<std::string>& files, uint8_t version = 1);
 
-  Block read();
+  Block& read();
   void process(int streamId, int concurrency=-1, int width=160, int height=160,
       const std::string& url="http://127.0.0.1:5000");
 
+  Block& currentBlock() { return m_curBlock; }
+
+  class iterator;
+  iterator begin() { return iterator(this); }
+  iterator end() { return iterator(nullptr); }
+
+  class iterator
+  {
+  public:
+    using self_type = iterator;
+    using value_type = Block;
+    using reference = Block&;
+    using pointer = Block*;
+    using iterator_category = std::input_iterator_tag;
```

Comment:
Looks like you aren't using this?

Reply:
Not personally, I was just trying to follow all of the requirements for making an input iterator, and I think this is one of them. If we pass it into an STL algorithm, it might use it. I could potentially use it by checking it in the `createSTEMImage` function to make sure it is an input_iterator_tag.

Reply:
Ok, thanks, I thought the tagging was done differently.